src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -480,4 +490,64 @@ private String getInvoiceLineNumberEndpoint(String id) {
 
   }
 
+  /**
+   * Updates the invoice's poNumbers field, following an invoice line creation, update or removal.
+   * @param invoice - the invoice of the modified invoice line
+   * @param invoiceLine - the modified invoice line
+   * @param invoiceLineFromStorage - the old version of the invoice line
+   * @param requestContext - used to start new requests
+   */
+  private CompletableFuture<Void> updateInvoicePoNumbers(Invoice invoice, InvoiceLine invoiceLine,
+      InvoiceLine invoiceLineFromStorage, RequestContext requestContext) {
+
+    if (invoiceLineFromStorage == null && invoiceLine.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    if (invoiceLine.getPoLineId() == null && invoiceLineFromStorage != null && invoiceLineFromStorage.getPoLineId() == null)
+      return CompletableFuture.completedFuture(null);
+    String poLineId = (invoiceLineFromStorage == null || invoiceLine.getPoLineId() != null) ? invoiceLine.getPoLineId() :
+      invoiceLineFromStorage.getPoLineId();
+    return orderService.getPoLine(poLineId, requestContext)
+      .thenCompose(poLine -> orderService.getOrder(poLine.getPurchaseOrderId(), requestContext))
+      .thenCompose(order -> {
+        List<String> invoicePoNumbers = invoice.getPoNumbers();
+        String orderPoNumber = order.getPoNumber();
+        if (orderPoNumber == null)
```

Comment:
`if (orderPoNumber == null)` - one more redundant check. Every created order should have ponumber field populated. If its not - it means that there is a data corruption and we shouldnt ignore it just by returning CompletableFuture.completedFuture(null). 
So, you can remove this check and leave this unhandled

Reply:
If I remove that I might have to revisit all the unit tests. Currently the old tests use a PO without a number, so poNumbers updates are not needed and the tests keep working without taking additional invoice updates into account. We might find another solution for that.
Another issue is that poNumber is not required by the schema, so we could potentially have older records without it.

Reply:
Could you create a method that will check if the numbers need to be updated?
1. Lines : 504, 506 -> isPoNumbersUpdateNeeded(invoiceLineFromStorage, invoiceLine) if return true then proceed else  CompletableFuture.completedFuture(null);
2. Line 516 -> isRemovePoNumberNeeded(invoiceLineFromStorage, invoiceLine)

Reply:
@grabsefx is it OK to leave `if (orderPoNumber == null)` , or do you still want me to remove it, and add a poNumber to all po mocks, and adjust all the tests that create or modify an invoice line and check the number of updates ?

Reply:
My thoughts: it is better to have an exception because of data corruption or inconsistency and stop the buisnes flow rather than ignore it. So it would be great to have mock data and unit tests be adjusted according to buisnes logic. As I said previously every created order will contain `poNumber` field populated. You can leave it as is. I think it could be done in scope of the separate story, as the tech debt.

Reply:
OK, I removed it and adjusted the tests. We should be aware, however, that most integration tests are using orders with no poNumber.

Reply:
Its ok. When you post order without poNumber - it will be generated automatically

Reply:
OK, I did not realize that.